_source/_posts/2019-12-26-java-mongodb-crud.md

```diff
@@ -0,0 +1,519 @@
+        },
+        "self": {
+            "href": "http://localhost:8080/fridge{?page,size,sort}",
+            "templated": true
+        }
+    },
+    "page": {
+        "number": 0,
+        "size": 20,
+        "totalElements": 3,
+        "totalPages": 1
+    }
+}
+
+```
+
+Notice that the returns JSON gives you the URL for each individual item. If you wanted to delete the first item in the list above, you could run the following request.
+
+```bash
+http DELETE :8080/fridge/5dae7b4c6a99f01364de916c
+```
+
+The long string,`5dae7b4c6a99f01364de916c`, is the unique ID for that item. MongoDB doesn't use sequential integer ID numbers like SQL databases often do. It uses randomly generated UUIDs instead.
+
+If you wanted to update an item, you could use a PUT, as shown below.
+
+```bash
+http PUT :8080/fridge/5dae7b4f6a99f01364de916e name="old pizza" expiration="03-30-2020" owner="Erin"
+```
+
+Note that with a PUT you have to send data for all the fields, not just the field you want to update, otherwise the omitted fields are set to null. If you just want to update select fields, use a PATCH.
+
+```bash
+http PATCH :8080/fridge/5dae7b4f6a99f01364de916e owner="Andrew"
+```
+
+With that rather paltry amount of work, you've created a MongoDB database model and exposed it to the world using a REST API. Pretty sweet!
+
+The next step is to secure it. The last thing you need is hackers breaking into your house and stealing your pizza and cheese.
+
+## Create an OIDC Application for Your Java + MongoDB App
+
+Okta is a software-as-service identity management provider. We provide solutions that make adding authentication and authorization to web applications easy. In this tutorial, you are going to use Okta to add JSON Web Token authentication and authorization to your application using OAuth 2.0 and OpenID Connect (OIDC). 
+
+OAuth 2.0 is an authorization protocol (verifying what the client or user is allowed to do) and OIDC is an authentication protocol (verifying the identity of the user) built on top of OAuth 2.0. They are a set of open standards that help ensure your web application's security is handled safely and effectively. Together they provide a complete authentication and authorization protocol. 
+
+They are not, however, implementations. That's where Okta comes in. Okta will be the identity provider and your Spring Boot app will be the client.
+
+You should have already signed up for a free developer account at Okta. Navigate to the developer dashboard at [https://developer.okta.com](https://developer.okta.com). If this is your first time logging in, you may need to click the **Admin** button.
+
+To configure JSON Web Token (JWT) authentication and authorization, you need to create an OIDC application. 
+
+From the top menu, click on the  **Applications** button. Click the  **Add Application**  button.
+
+Select application type  **Web** and click  **Next**.
+
+Give the app a name. I named mine "Spring Boot Mongo".
```

Comment:
Was this a list? It looks fine either way IMHO

Reply:
No, this is the way Andrew writes his instructions.